test/app.test.ts

```diff
@@ -2,85 +2,140 @@ import * as fs from 'fs';
 import * as os from 'os';
 import * as path from 'path';
 import { Node, Construct } from 'constructs';
-import { Testing, Chart, App, ApiObject } from '../src';
+import { Testing, Chart, App, ApiObject, AppProps, YamlOutputType } from '../src';
+
 
 test('empty app emits no files', () => {
-  // GIVEN
-  const app = Testing.app();
+  const outputTypes: Array<AppProps> = [
+    { yamlOutputType: YamlOutputType.FILE_PER_CHART },
+    { yamlOutputType: YamlOutputType.FILE_PER_APP },
+    { yamlOutputType: YamlOutputType.FILE_PER_RESOURCE },
+  ];
+  for (const props of outputTypes) {
+    // GIVEN
+    const app = Testing.app(props);
 
-  // WHEN
-  app.synth();
+    // WHEN
+    app.synth();
 
-  // THEN
-  expect(fs.readdirSync(app.outdir)).toHaveLength(0);
+    // THEN
+    expect(fs.readdirSync(app.outdir)).toHaveLength(0);
+  }
 });
 
 test('app with two charts', () => {
-  // GIVEN
-  const app = Testing.app();
+  const testSpecs = [
+    {
+      props: { yamlOutputType: YamlOutputType.FILE_PER_CHART },
+      result: ['chart1.k8s.yaml', 'chart2.k8s.yaml'],
+    },
+    {
+      props: { yamlOutputType: YamlOutputType.FILE_PER_APP },
+      result: [],
+    },
```

Comment:
I think we should be consistent across options (so the FILE_PER_APP option should always generate a file even if it's empty, since that matches the existing behavior of FILE_PER_CHART).